nucypher/cli/commands/bond.py

```diff
@@ -0,0 +1,197 @@
+        emitter.message(ALREADY_BONDED.format(provider=onchain_staking_provider, operator=operator_address), color='red')
+        raise click.Abort()  # dont steal bananas
+
+    # Check that operator is not human
+    if staking_provider != operator_address:
+        # if the operator has a beneficiary it is the staking provider.
+        beneficiary = agent.get_beneficiary(staking_provider=operator_address)
+        if beneficiary != NULL_ADDRESS:
+            emitter.message(UNEXPECTED_HUMAN_OPERATOR, color='red')
+            raise click.Abort()
+
+    #
+    # Bond
+    #
+
+    if not force:
+        click.confirm(CONFIRM_BONDING.format(provider=staking_provider, operator=operator_address), abort=True)
+    transacting_power.unlock(password=get_client_password(checksum_address=staking_provider, envvar=NUCYPHER_ENVVAR_STAKING_PROVIDER_ETH_PASSWORD))
+    emitter.echo(BONDING.format(operator=operator_address))
+    receipt = agent.bond_operator(operator=operator_address, transacting_power=transacting_power, provider=staking_provider)
+    paint_receipt_summary(receipt=receipt, emitter=emitter)
+
+
+@click.command('unbond')
+@option_registry_filepath
+@option_provider_uri(required=True)
+@option_signer_uri
+@option_staking_provider
+@option_network()
+@option_force
+def unbond(registry_filepath, provider_uri, signer_uri, staking_provider, network, force):
+    """Unbonds an operator from an authorized staking provider."""
+
+    #
+    # Setup
+    #
+    
+    emitter = StdoutEmitter()
+    if not signer_uri:
+        emitter.message('--signer is required', color='red')
+        raise click.Abort()
+    if not network:
+        network = select_network(emitter=emitter)
+
+    connect_to_blockchain(provider_uri=provider_uri, emitter=emitter)
+    registry = get_registry(network=network, registry_filepath=registry_filepath)
+    agent = ContractAgency.get_agent(PREApplicationAgent, registry=registry)
+    signer = Signer.from_signer_uri(signer_uri)
+    transacting_power = TransactingPower(account=staking_provider, signer=signer)
+
+    #
+    # Check
+    #
+    
+    bonded, onchain_operator_address = is_bonded(agent=agent, staking_provider=staking_provider, return_address=True)
+    if not bonded:
+        emitter.message(NOT_BONDED.format(provider=staking_provider), color='red')
+        raise click.Abort()
+    time_elapsed(emitter=emitter, agent=agent, staking_provider=staking_provider)
+
+    #
+    # Unbond
+    #
+
+    if not force:
+        click.confirm(CONFIRM_UNBONDING.format(provider=staking_provider, operator=onchain_operator_address), abort=True)
+    transacting_power.unlock(password=get_client_password(checksum_address=staking_provider, envvar=NUCYPHER_ENVVAR_STAKING_PROVIDER_ETH_PASSWORD))
+    emitter.echo(UNBONDING.format(operator=onchain_operator_address))
+    receipt = agent.bond_operator(operator=NULL_ADDRESS, transacting_power=transacting_power, provider=staking_provider)
```

Comment:
Should `bond_operator()` take `staking_provider` as a keyword-arg instead of `provider`? Other agent functions seem to also take `staking_provider`.

Reply:
✔️